test/e2e/ingress/route.go

```diff
@@ -0,0 +1,54 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package ingress
+
+import (
+	"github.com/api7/ingress-controller/test/e2e/scaffold"
+	"github.com/onsi/ginkgo"
+	"github.com/stretchr/testify/assert"
+	"time"
+)
+
+var _ = ginkgo.Describe("upstream expansion", func() {
+	s := scaffold.NewDefaultScaffold()
+	ginkgo.It("create and then scale to 2 ", func() {
+		apisixRoute := `
+apiVersion: apisix.apache.org/v1
+kind: ApisixRoute
+metadata:
+  name: httpbin-route
+spec:
+  rules:
+  - host: httpbin.com
+    http:
+      paths:
+      - backend:
+          serviceName: httpbin-service-e2e-test
+          servicePort: 80
+        path: /ip
+`
+		s.CreateApisixRouteByString(apisixRoute)
+
+		err := s.EnsureNumApisixRoutesCreated(1)
+		assert.Nil(ginkgo.GinkgoT(), err, "Checking number of routes")
+		err = s.EnsureNumApisixUpstreamsCreated(1)
+		assert.Nil(ginkgo.GinkgoT(), err, "Checking number of upstreams")
+		s.ScaleHTTPBIN(2)
+		time.Sleep(25 * time.Second)
```

Comment:
Why sleeping 25s? If we need to wait the service to ready, just poll it.

Reply:
Done